src/esm_runscripts/oasis.py

```diff
@@ -465,9 +465,24 @@ def add_restart_files(self, restart_file, fconfig):
                 # folder of the parent experiment to be branched off from:
                 glob_search_file = f"{config['ini_restart_dir']}{restart_file}_????????-{config['ini_restart_date'].year}{config['ini_restart_date'].month:02}{config['ini_restart_date'].day:02}"
                 branchoff_restart_file = glob.glob(glob_search_file)
-                # Note: What if there are more than one found?
+                branchoff_restart_file.sort()
                 if branchoff_restart_file:
-                    branchoff_restart_file = os.path.basename(branchoff_restart_file[0])
+                    # Note: What if there are more than one found? Let the user decide which one to take.
+                    if len(branchoff_restart_file) == 1:
+                        branchoff_restart_file = os.path.basename(branchoff_restart_file[0])
+                    else:
+                        if not gconfig["isinteractive"]:
+                            # If more than one restart file found that matches ini_restart_date,
+                            # ask the user to select from the result list:
+                            questions = [
+                                {
+                                    "type": "select",
+                                    "name": "OASIS restart file",
+                                    "message": "More than one OASIS restart file was found for your branchoff experiment that matches the ini_restart_date you selected. Please select one of the following OASIS restart files:",
+                                    "choices": branchoff_restart_file,
+                                },
+                            ]
+                            branchoff_restart_file = os.path.basename(questionary.prompt(questions)["OASIS restart file"])
```

Comment:
I'm not entirely sure from memory, but I think you need `.ask()` still here:
```suggestion
                            branchoff_restart_file = os.path.basename(questionary.prompt(questions).ask()["OASIS restart file"])
```

Reply:
This was correct. But I changed it a bit and use questionary.form now. This makes it a few lines shorter.